src/employee/employee.service.ts

```diff
@@ -0,0 +1,51 @@
+import { IEmployeeRepository } from "./employee.repository";
+import { EmployeeType, EmployeeCreateType } from "./interfaces";
+import { IRepositories } from "../repositories";
+
+export interface IEmployeeService {
+  createEmployee: (employee: EmployeeCreateType) => Promise<EmployeeType>;
+  getAllEmployees: () => Promise<EmployeeType[]>;
+  getEmployeeById: (id: number) => Promise<EmployeeType | undefined>;
+}
+
+class EmployeeService implements IEmployeeService {
+  private readonly employeeRepository: IEmployeeRepository;
+  constructor({ employeeRepository }: IRepositories) {
+    this.employeeRepository = employeeRepository;
+  }
+
+  async createEmployee({
+    name,
+    surname,
+    githubAccount,
+    employmentType,
+  }: EmployeeCreateType) {
+    const user = await this.employeeRepository.findUnique(
+      "githubAccount",
+      githubAccount
+    );
+
+    if (user) {
+      throw new Error(
```

Comment:
Przydałby ci się jakiś wyjątek domenowy.

Reply:
dodałbym to do następnego brancha